File: 0409_Longest Palindrome/LongestPalindrome.cs
public class Solution
{
    public int LongestPalindrome(string s)
    {
        if (s == null || s.Length == 0) return 0;
        var set = new HashSet<char>();
        int ans = 0;
        foreach (var c in s)
        {
            if (set.Contains(c))
            {
                set.Remove(c);
                ans += 2;
            }
            else
            {
                set.Add(c);
            }
        }

        return set.Count > 0 ? ans + 1 : ans;
    }
}

File: 0409_Longest Palindrome/LongestPalindrome.java
class Solution {
    public int longestPalindrome(String s) {
        int[] arr = new int[128];
        for(char c : s.toCharArray()) {
            arr[c]++;
        }

        int ans = 0;
        boolean hasOdd = false;
        for(int v : arr) {
            if(v% 2 == 0) {
                ans+=v;
            }else if(v > 0){
                ans += (v-1);
                hasOdd = true;
            }
        }

        if(hasOdd) ans++;
        return ans;
    }
}